docs/deployment/microbit-deployment-draft.md

```diff
@@ -0,0 +1,166 @@
+# Python deployment on python.microbit.org
+
+This spec outlines how python.microbit.org will become the primary web-hosted editor for Micropython micro:bit programming, deprecating what is at microbit.co.uk. 
+
+It also includes a mechanism to make the same editor available to users offline, so that they can have a familiar environment when working without an internet connection.
+
+## Versioning
+
+In order to balance the requirements of stability required by teachers and people producing printed materials and books, with a desire to ensure the latest features are available to advanced/curious users the following mechanisms will be in place:
+
+From the top level version, it is highly desirable (I’d like to say mandatory?) that the top level version deterministically identify the precise version of the underlying components that will run on the micro:bit (in reality, this means the MicroPython version/build which should itself point to a DAL version and then an mbed one from there)
+
+Not only could this enable direct viewing of dependent source (including the underlying C/C++), it could offer in some editors a list of known bugs in that version, and url links to the version that fixed it.
+
+### Semantic versioning
+
+The editor will use semantic versioning (http://semver.org/). In addition to breaking API changes, the following things will be considered ‘MAJOR’ changes
+1) Addition of significant features if they interact with the existing deployed features
+2) Things that would change screenshots of the editor that are published online or in books.
+
+Additional buttons on a toolbar are fine, but removing or renaming existing buttons should be considered a breaking change.
+
+Anything that could stop a script that previously worked from working must be considered a breaking change. In the case of MicroPython this means that things increasing memory consumption or reducing the flash available for a users’ script are likely breaking changes.
+
+{<< what other things should constitute a breaking change, and should we allow certain kinds of breaking change inside a version of the editor? IE can we support more than one MicroPython version for a particular deployed version of the front-end?>>}}
+
+It may be desirable to agree a high water mark in memory and growing beyond this is considered a breaking change. It would then be necessary to police memory growth but people do already do this I think. This would allow small increases in memory usage to be possible without a breaking change.
+
+### URL versioning scheme
+
+For each published major version, the deployed editor will live at
+
+`python.microbit.org/v<major version>/editor.html`
+
+Because the use of semantic versioning ensures that things will not break for the user when upgrading minor versions, there will not be fine-grained control over which minor version a user gets. 
```

Comment:
Agree with Carlos. v2 needs to be the "don't make me think" version, and resources should always use that form of URL and 'share' should always create that form of URL. The minor versioned ones should be available but 'under the hood' - does that work for you both?